torch/_functorch/partitioners.py

```diff
@@ -837,15 +843,33 @@ def get_node_weight(node) -> int:
         # Heuristic to bias towards nodes closer to the backwards pass
         # Complete guess about current value
         mem_sz = int(mem_sz * (1.1 ** max(min(node.dist_from_bw, 100), 1)))
-        # mem_sz = int(mem_sz + node.dist_from_bw)
-
         if is_materialized(node):
             return mem_sz
         else:
             return mem_sz * 2
 
     nx_graph = nx.DiGraph()
-    for node in full_bw_graph.nodes:
+    banned_nodes = set()
+
+    def ban_recomputation_if_allowed(node):
+        # This bans recomputation of the node unless we've been forced not to by
+        # user annotation
+        # NB: "recompute" == 0 means it's not banned from recomputing
```

Comment:
Not sure about this. We marked the output nodes of AC blocks with "recompute" == 0. Shouldn't we ban recomputation for those nodes?

Reply:
This was correct (in some sense), forward fix here: https://github.com/pytorch/pytorch/pull/122841